### PR TITLE
Update to Nibbles 4.0.4

### PR DESCRIPTION
**Pull request by yakushabb**
Update to Nibble 4.0.4. JSON copied from the development repository and adjusted as necessary.

### CODE REVIEW

**Comment by ghost** [comment]
@naipotato @nedrichards FYı

**Comment by yakushabb** [comment]
@naipotato @nedrichards FYI

File: org.gnome.Nibbles.json [hunk 1]
```diff
@@ -1,47 +1,76 @@
...
+        "/share/pkgconfig",
+        "/share/vala",
+        "*.la",
+        "*.a"
+    ],
     "modules": [
         {
             "name" : "gsound",
-            "buildsystem": "meson",
+            "buildsystem" : "meson",
```
Comment (bbhtt) [inline org.gnome.Nibbles.json]:
```suggestion
            "buildsystem": "meson",
```

File: org.gnome.Nibbles.json [hunk 2]
```diff
@@ -1,47 +1,76 @@
...
             "sources" : [
                 {
-                    "type" : "archive",
-                    "url" : "https://download.gnome.org/sources/gsound/1.0/gsound-1.0.3.tar.xz",
-                    "sha256": "ca2d039e1ebd148647017a7f548862350bc9af01986d39f10cfdc8e95f07881a"
+                    "type" : "git",
+                    "url" : "https://gitlab.gnome.org/GNOME/gsound.git",
+                    "tag": "1.0.3"
```
Comment (bbhtt) [inline org.gnome.Nibbles.json]: Can you please keep this to the archive? Is there any reason to prefer the git? I'd avoid stylistic changes to minimise the diff.

File: org.gnome.Nibbles.json [hunk 3]
```diff
@@ -1,47 +1,76 @@
...
                 {
-                    "name" : "libcanberra",
-                    "sources" : [
-                        {
-                            "type" : "archive",
-                            "url" : "http://0pointer.de/lennart/projects/libcanberra/libcanberra-0.30.tar.xz",
-                            "sha256" : "c2b671e67e0c288a69fc33dc1b6f1b534d07882c2aceed37004bf48c601afa72"
-                        }
-                    ],
-                    "config-opts" : [
```
Comment (bbhtt) [inline org.gnome.Nibbles.json:40]: Same here

File: org.gnome.Nibbles.json [hunk 4]
```diff
@@ -50,9 +79,9 @@
             "buildsystem": "meson",
             "sources": [
                 {
-                    "type": "archive",
-                    "url": "https://download.gnome.org/sources/libgnome-games-support/1.8/libgnome-games-support-1.8.2.tar.xz",
-                    "sha256": "28434604a7b038731ac0231731388ff104f565bb2330cc24e78cda04cfd3ef7d"
+                    "type": "git",
+                    "url": "https://gitlab.gnome.org/GNOME/libgnome-games-support.git",
```
Comment (bbhtt) [inline org.gnome.Nibbles.json]: Same here

2.0 has an archive https://download.gnome.org/sources/libgnome-games-support/2.0/

File: org.gnome.Nibbles.json [hunk 5]
```diff
@@ -1,47 +1,76 @@
...
-                        {
-                            "type" : "archive",
-                            "url" : "http://0pointer.de/lennart/projects/libcanberra/libcanberra-0.30.tar.xz",
-                            "sha256" : "c2b671e67e0c288a69fc33dc1b6f1b534d07882c2aceed37004bf48c601afa72"
-                        }
-                    ],
-                    "config-opts" : [
+                    "name": "libcanberra",
+                    "config-opts": [
                         "--disable-alsa",
```
Comment (bbhtt) [inline org.gnome.Nibbles.json]: Same here, I'd avoid the reorganisation to minimise the diff.

File: org.gnome.Nibbles.json (hunk 2 above)
Comment (ghost) [reply org.gnome.Nibbles.json]: I started to prefer XZ incident. But not a problem for me.

Comment (bbhtt) [reply org.gnome.Nibbles.json]: GNOME is fine because repo access goes through the Gitlab admins and there is a vouching system.

**Pull request merged**